### PR TITLE
Fixing Attachment Error New Fork

### PR DESCRIPTION
**Pull request by Gamma169**
From PR https://github.com/go-gomail/gomail/pull/97

### CODE REVIEW

File: message_test.go [hunk 1]
```diff
@@ -256,6 +257,9 @@ func TestAttachmentOnly(t *testing.T) {
 	}
 
 	testMessage(t, m, 0, want)
+	if err := teardownFile("/tmp/test.pdf"); err != nil {
+		panic(err)
```
Comment (ivy) [inline message_test.go:261]: It's generally not a good practice to panic unless absolutely necessary. Have a look at using `t.Errorf("teardown: %s", err)` instead.

**Comment by ivy** [comment]
@Gamma169 Thanks for making time for this over your weekend. On further consideration, I don't think we'll be able to benefit from this improvement without sacrificing usability/features somewhere else:

* [`SetCopyFunc`](https://godoc.org/gopkg.in/mail.v2#SetCopyFunc) allows replacing the underlying file-backed copier with anything that acts on an `io.Writer` (as you see in the mocked tests). We need something that still allows this functionality as it's useful in other areas outside of testing such as in go-gomail/gomail#78.
* [`Rename`](https://godoc.org/gopkg.in/mail.v2#Rename) makes things trickier as we also can't rely on the filename matching what's on disk.

I was considering simply adding an `error` return value to [`Embed`](https://godoc.org/gopkg.in/mail.v2#Embed) and [`Attach`](https://godoc.org/gopkg.in/mail.v2#Attach) but that also breaks `SetCopyFunc` for non-file-backed attachments. To sum things up, we're going to have to wait on this until there's a solid plan for `v3`.

**Comment by pedromorgan** [comment]
@Ivy why?

this is a good patch, got a lot of stuff in there..

at least create the v3 branch..

**Comment by pedromorgan** [comment]
@ivy serious do not reject.. accept everything as a possibility..

and id u do that.. then u have a whole heap behind u,,

Thing u a good "leader" in making noise..
But this is code ok...
and its differrent kinda respect.....and trust

**Comment by Gamma169** [comment]
@ivy Understood about the `SetCopyFunc` function.  @pedromorgan The problem is with that function, a person can attach data and label it as a file that doesn't exist.  Thus triggering the error on the send, when there actually is no file.

I can think about things a little more and figure out a way to get rid of the bug (because I think it's pretty egregious).

**Pull request closed without merging**